fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -954,6 +958,57 @@ public Void visitLikePredicate(LikePredicate node, Scope scope) {
             return null;
         }
 
+        @Override
+        public Void visitMatchExpr(MatchExpr node, Scope scope) {
+            Type type1 = node.getChild(0).getType();
+            Type type2 = node.getChild(1).getType();
+
+            if (!type1.isStringType() && !type1.isNull()) {
+                throw new SemanticException("left operand of MATCH must be of type STRING");
+            }
+
+            if (!(node.getChild(0) instanceof SlotRef)) {
+                throw new SemanticException("left operand of MATCH must be column ref");
+            }
+
+            for (Field f : scope.getRelationFields().getAllFields()) {
```

Comment:
please add some tests for query like `select * from (select t1.cola as colb from t1 join t2) t1 where t1.colb match 'abc'`. Is the query like this can use GIN?

Reply:
Updated, remove the check code here. BE will return error if the column using match has no gin